tools/run_tests/python_utils/check_on_pr.py

```diff
@@ -46,13 +49,28 @@ def _jwt_token():
 def _access_token():
     global _ACCESS_TOKEN_CACHE
     if _ACCESS_TOKEN_CACHE == None or _ACCESS_TOKEN_CACHE['exp'] < time.time():
-        resp = requests.post(
-            url='https://api.github.com/app/installations/%s/access_tokens' %
-            _INSTALLATION_ID,
-            headers={
-                'Authorization': 'Bearer %s' % _jwt_token().decode('ASCII'),
-                'Accept': 'application/vnd.github.machine-man-preview+json',
-            })
+        for i in range(_ACCESS_TOKEN_FETCH_RETRIES):
+            resp = requests.post(
+                url='https://api.github.com/app/installations/%s/access_tokens'
+                % _INSTALLATION_ID,
+                headers={
+                    'Authorization': 'Bearer %s' % _jwt_token().decode('ASCII'),
+                    'Accept': 'application/vnd.github.machine-man-preview+json',
+                })
+            if resp.status_code == 200:
+                break
+            else:
+                print("Fetch access token from Github API failed:")
+                print(resp.json())
+                if i != _ACCESS_TOKEN_FETCH_RETRIES - 1:
+                    print('Retrying after %.2f second.' %
+                          _ACCESS_TOKEN_FETCH_RETRIES_INTERVAL_S)
+                    time.sleep(_ACCESS_TOKEN_FETCH_RETRIES_INTERVAL_S)
+
+        if resp.status_code != 200:
```

Comment:
This could be an `else` block hanging off of the `for`.

Reply:
Updated. That's Pythonic!